operator/main.py

```diff
@@ -464,22 +466,203 @@ def handle_modified(core_v1_client, obj):
+
+    return storages
+
+def set_num_pvs(storages, volname):
+    """Set PV count for all listed storages"""
+
+    for storage in storages:
 
-def handle_deleted():
+        if storage.storage_name != volname:
+            continue
+
+        dbpath = "/bricks/" + storage.storage_name + "/data/brick/stat.db"
+        query = ("select count(pvname) from pv_stats;")
+        cmd = ["kubectl", "exec", "-i",
+                storage.storage_units[0].podname,
+                "-c", "glusterfsd", "-nkadalu", "--", "sqlite3",
+                dbpath,
+                query
+        ]
+
+        try:
+            resp = utils.execute(cmd)
+            parts = resp.stdout.strip().split("\n")
+            storage.total_size_bytes = float(parts[0].strip())
+            pv_stats_parts = parts[1].strip().split("|")
+            storage.pv_count = int(pv_stats_parts[0])
+        except utils.CommandError as err:
+            print("Failed to get size details of the "
+                  "storage \"%s\"" % storage.storage_name,
+                  file=sys.stderr)
+            print(err, file=sys.stderr)
+            print()
+        except FileNotFoundError:
+            # remove 'args' : args not defined
+            utils.kubectl_cmd_help(args.kubectl_cmd)
+
+def handle_deleted(core_v1_client, obj):
     """
     If number of pvs provisioned from that volume
     is zero - Delete the respective server pods
     If number of pvs is not zero, wait or periodically
     check for num_pvs. Delete Server pods only when pvs becomes zero.
     """
     # TODO
+    logging.info(logf(
+        "Inside handle_deleted()"
+    ))
+
+    volname = obj["metadata"]["name"]
+
     logging.warning(logf(
-        "DELETED handle called, but not implemented"
+        "Delete {} requested".format(volname)
     ))
 
+    cmd = ["kubectl", "get", "configmap", "kadalu-info", "-nkadalu", "-ojson"]
+
+    try:
+        resp = utils.execute(cmd)
+        storages = storage_list.list_storages(resp.stdout, args)
+    except utils.CommandError as err:
+        utils.command_error(cmd, err.stderr)
+    except FileNotFoundError:
+        utils.kubectl_cmd_help(args.kubectl_cmd)
+
+    set_num_pvs(storages, volname)
+
+    for storage in storages:
+        if storage.storage_name == volname and storage.pv_count != 0:
+            logging.warning(logf(
+                "Number of PVs from {} is not zero".format(volname)
+            ))
+            # TODOLIST : Implement is_num_pv_zero()
+            return
+        elif storage.storage_name == volname and storage.pv_count == 0:
+            delete_config_map(core_v1_client, obj)
+            delete_server_pods(obj)
+
+            filename = os.path.join(MANIFESTS_DIR, "services.yaml")
+            template(filename, namespace=NAMESPACE, volname=volname)
+            lib_execute(KUBECTL_CMD, DELETE_CMD, "-f", filename)
+            logging.info(logf("Deleted Service", volname=volname, manifest=filename))
+
```

Comment:
Break after this.

It should be fine to iterate all in this case, but with the `break` you can come out of loop early.